src/Connectors/src/ConnectorBase/ConnectionTypeLocator.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Steeltoe.Connector.CosmosDb;
+using Steeltoe.Connector.MongoDb;
+using Steeltoe.Connector.MySql;
+using Steeltoe.Connector.Oracle;
+using Steeltoe.Connector.PostgreSql;
+using Steeltoe.Connector.RabbitMQ;
+using Steeltoe.Connector.Redis;
+using Steeltoe.Connector.Services;
+using Steeltoe.Connector.SqlServer;
+using System;
+
+namespace Steeltoe.Connector
+{
+    internal static class ConnectionTypeLocator
+    {
+        internal static Type GetConnectionInfoType(string input)
+        {
+            return true switch
+            {
+                bool _ when input.Equals("cosmosdb", StringComparison.InvariantCultureIgnoreCase) => typeof(CosmosDbConnectionInfo),
```

Comment:
This feels wrong, but hard to put finger on it.  E.g., the fact that "mysql" relates to `MySqlConnectionInfo` should be encapsulated in the `MySqlConnectionInfo` class (or related if using a factory of some sort).

So something more along lines of:
```
for info in connection infos:
  if connection info accepts input:
    return connection info
throw exception
```

Reply:
But even knowing the connection classes seems wrong as well.  Could the connection classes register them selves upon init?  Otherwise wouldn't this code need to be updated when a new connection type is added?